utils/get_prometheus_data.py

```diff
@@ -0,0 +1,145 @@
+            return results
+        except Exception as e:
+            logger.warn("failure to get labels %s" % e)
+
+    def get_all_metrics(self):
+
+        # check get_data bool, if false by-pass all processing
+        if self.get_data:
+            start_time = time.time()
+
+            filename = os.path.join(sys.path[0], 'utils/prometheus_labels/included_labels.json')
+            with open(filename, 'r') as f:
+                datastore = json.load(f)
+
+            for label in self.get_label_list():
+
+                if label in datastore["data"]:
+
+                    query = "/api/v1/query_range?query=" + label
+                    query = query + "&start=" + self.start
+                    query = query + "&end=" + self.end
+                    query = query + "&step=" + str(self.T_Delta) + "s"
+                    # query_start_time = time.time()
+                    try:
+                        response = self.api_call(query)
+                    except Exception as e:
+                        logger.warn("failure to get metric results %s" % e)
+
+                    results = response['result']
+
+                    # results is a list of all hits
+
+                    for result in results:
+                        # clean up name key from __name__ to name
+                        result["metric"]["name"] = ""
+                        result["metric"]["name"] = result["metric"]["__name__"]
+                        del result["metric"]["__name__"]
+                        ROC_V1 = 0
+                        # each result has a list, we must flatten it out in order to send to ES
+                        for value in result["values"]:
+                            # fist index is time stamp
+                            timestamp = datetime.utcfromtimestamp(value[0]).strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+                            # second index is value of metric
+                            if "NaN" in value[1]:  # need to handle values that are NaN, Inf, or -Inf
+                                metric_value = 0
+                            else:
+                                """
+                                TODO add logic for calculating the rate of increase per second
+                                prometheus will sometimes reset counters so we need to detect is V2 is 
+                                less than V1 and if add V2 to V1 divided by T-delta (30s). we should 
+                                set the T_delta or time step as a class varable just in case we change
+                                the value later the calculation is still the same.
+                                """
+                                metric_value = float(value[1])
+                                # have to use float but in python3 need to use int()
+
+                                if ROC_V1 == 0:
+                                    ROC_V1 = metric_value
+                                    ROC = 0
```

Comment:
since ROC is what is included in flat_doc below, why is it always set to 0?   what use is ROC_V1 and metric_value if ROC_V1 == 0?   Is this the case where the metric is undefined?  is this really the right value to return in this circumstance?  I would think you would either just not return anything or return the most recent known  value of the counter.

Reply:
This covers the initial metric value when we can not calculate a rate of change.

Reply:
so here's the problem with that.   If you guess that the value is zero and it's really very high to begin with, you will calculate a very large rate of change that is not in fact accurate.     I think it's safer to not try to compute a rate of change unless you have the value for the previous time and the current time.  Make sense?

Reply:
OBE